packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -303,34 +334,35 @@ export const prepareStatusManager = (
       },
 
       /**
-       * Remove and return an `ADVANCED` or `OBSERVED` tx waiting to be `SETTLED`.
+       * Remove and return the oldest pending settlement transaction that matches the given
+       * forwarding account and amount. Since multiple pending transactions may exist with
+       * identical (account, amount) pairs, we process them in FIFO order.
        *
        * @param {NobleAddress} nfa
        * @param {bigint} amount
-       * @returns {Pick<PendingTx, 'status' | 'txHash'> | undefined} undefined if nothing
-       *   with this address and amount has been marked pending.
+       * @returns {Pick<PendingTx, 'status' | 'txHash'> | undefined} undefined if no pending
+       *   transactions exist for this address and amount combination.
        */
       dequeueStatus(nfa, amount) {
         const key = makePendingTxKey(nfa, amount);
-        if (!pendingTxs.has(key)) return undefined;
-        const pending = pendingTxs.get(key);
+        if (!pendingSettleTxs.has(key)) return undefined;
+        const pending = pendingSettleTxs.get(key);
 
-        const dequeueIdx = pending.findIndex(
-          x => x.status !== PendingTxStatus.Advancing,
-        );
-        if (dequeueIdx < 0) return undefined;
+        if (pending.length === 0) {
+          return undefined;
+        }
+        // extract first item
+        const { status, txHash } = pending[0];
 
+        // Update the pendingSettleTxs map
         if (pending.length > 1) {
           const pendingCopy = [...pending];
-          pendingCopy.splice(dequeueIdx, 1);
-          pendingTxs.set(key, harden(pendingCopy));
+          pendingCopy.shift(); // Remove first element
+          pendingSettleTxs.set(key, harden(pendingCopy));
```

Comment:
we shouldn't need to mutate to select elements from an array.
```suggestion
          const [_, ...remaining] = pending;
          pendingSettleTxs.set(key, harden(remaining));
```

Reply:
Agree this is awkward, but more so mutating when already we already made a copy. Updated.